hnn_core/drives.py

```diff
@@ -0,0 +1,67 @@
+"""External drives to network."""
+
+# Authors: Christopher Bailey <bailey.cj@gmail.com>
+
+
+def _get_target_populations(weights_ampa, weights_nmda):
+    # allow passing weights as None, but make iterable here
+    if weights_ampa is None:
+        weights_ampa = dict()
+    if weights_nmda is None:
+        weights_nmda = dict()
+    target_populations = (set(weights_ampa.keys()) | set(weights_nmda.keys()))
+    return target_populations, weights_ampa, weights_nmda
+
+
+def _check_drive_parameter_values(drive_type, **kwargs):
+    if 'sigma' in kwargs:
+        if kwargs['sigma'] < 0.:
+            raise ValueError('Standard deviation cannot be negative')
+    if 'numspikes' in kwargs:
+        if not kwargs['numspikes'] > 0:
+            raise ValueError('Number of spikes must be greater than zero')
+    if 'tstart' in kwargs:
+        if kwargs['tstart'] < 0:
+            raise ValueError(f'Start time of {drive_type} drive cannot be '
+                             'negative')
+    if 'tstop' in kwargs:
+        if kwargs['tstop'] < 0.:
+            raise ValueError(f'End time of {drive_type} drive cannot be '
+                             'negative')
+    if 'tstop' in kwargs and 'sim_end_time' in kwargs:
+        if kwargs['tstop'] > kwargs['sim_end_time']:
+            raise ValueError(f"End time of {drive_type} drive cannot exceed "
```

Comment:
Does the code break if this condition is not met? If not it seems a bit overly restrictive.

Reply:
I see your point, but is there a case where this would be valid? I guess I'm thinking it's better raise and error, since the user might be asking for something inconsistent? Not strongly for or against, are you?

Reply:
No strong feelings either so may as well keep it. I think regular use of the API will reveal if it's a nuisance or not.

Reply:
OK, keeping for now.